emrichen/template.py

```diff
@@ -24,7 +25,11 @@ def render(self, context, format='yaml'):
 
     @classmethod
     def parse(cls, data, format='yaml'):
-        return cls(template=parse(data, format=format))
+        filename = None
```

Comment:
You can replace all three lines with a simple `filename = (getattr(data, 'name', None) or None)`.